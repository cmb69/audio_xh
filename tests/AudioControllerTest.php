<?php

namespace Audio;

use ApprovalTests\Approvals;
use Audio\Infra\CsvFile;
use Audio\Model\MetaRepo;
use org\bovigo\vfs\vfsStream;
use PHPUnit\Framework\TestCase;
use Plib\View;

class AudioControllerTest extends TestCase
{
    public function testRendersAudioWidget(): void
    {
        $description = <<<'EOS'
        The Goldberg Variations (German: Goldberg-Variationen), BWV 988, is a musical composition
        for keyboard by Johann Sebastian Bach, consisting of an aria and a set of 30 variations.
        First published in 1741, it is named after Johann Gottlieb Goldberg,
        who may also have been the first performer of the work.
        EOS;
        vfsStream::setup("root");
        mkdir(vfsStream::url("root/userfiles/media"), 0777, true);
        touch(vfsStream::url("root/userfiles/media/goldberg.mp3"));
        touch(vfsStream::url("root/userfiles/media/goldberg.ogg"));
        mkdir(vfsStream::url("root/content"));
        file_put_contents(vfsStream::url("root/content/audio.csv"), "goldberg,Goldberg Variations,\"$description\"");
        $sut = new AudioController(
            vfsStream::url("root/userfiles/media/"),
            new MetaRepo(vfsStream::url("root/content/")),
            new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["audio"])
        );
        Approvals::verifyHtml($sut->defaultAction("goldberg", false, false));
    }
}
