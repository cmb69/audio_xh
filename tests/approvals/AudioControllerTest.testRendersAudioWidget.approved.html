
<figure itemscope itemtype="http://schema.org/AudioObject">
  <meta itemprop="name" content="Goldberg Variations">
  <meta itemprop="description" content="The Goldberg Variations (German: Goldberg-Variationen), BWV 988, is a musical composition
for keyboard by Johann Sebastian Bach, consisting of an aria and a set of 30 variations.
First published in 1741, it is named after Johann Gottlieb Goldberg,
who may also have been the first performer of the work.">
  <figcaption>Goldberg Variations</figcaption>
  <audio controls="controls" title="Goldberg Variations"  >
    <source src="vfs://root/userfiles/media/goldberg.ogg" type="audio/ogg; codecs=vorbis">
    <source src="vfs://root/userfiles/media/goldberg.mp3" type="audio/mpeg">
    <a href="vfs://root/userfiles/media/goldberg.mp3">Goldberg Variations</a>
  </audio>
</figure>
